bahmnicore-omod/src/main/resources/liquibase.xml

```diff
@@ -4578,134 +4578,148 @@
         </sql>
     </changeSet>
 
-    <changeSet id="bahmni-emrapi-202208101339" author="BAHMNI" >
+
+    <changeSet id="bahmni-emrapi-202208101439" author="BAHMNI" >
         <preConditions onFail="MARK_RAN">
             <sqlCheck expectedResult="1">
                 select count(*) from concept c
                 INNER join concept_name cn
-                on cn.concept_id=c.concept_id and cn.name='Visit Diagnoses' and cn.concept_name_type='FULLY_SPECIFIED' and cn.voided=0 and c.version is null;
+                on cn.concept_id=c.concept_id and cn.name='Visit Diagnoses' and cn.concept_name_type='FULLY_SPECIFIED' and cn.voided=0 and cn.locale = 'en' and c.version is null;
             </sqlCheck>
             <sqlCheck expectedResult="1">
+                <!-- The below query returns the concept which has only one concept reference mapping !-->
                 select count(*) from concept_reference_map rmap, concept_name cn where cn.concept_id = rmap.concept_id and cn.name='Visit Diagnoses'
-                and cn.concept_name_type='FULLY_SPECIFIED' and cn.voided=0 group by rmap.concept_id having count(*) = 1;
+                and cn.concept_name_type='FULLY_SPECIFIED' and cn.voided=0 and cn.locale = 'en' group by rmap.concept_id having count(*) = 1;
             </sqlCheck>
         </preConditions>
         <comment>Update version for the 'Visit Diagnoses' concept because Bahmni concept can be uniquely recognised and doens’t clash with CIEL, or other concepts</comment>
         <sql>
             SET @concept_id = 0;
 
             SELECT rmap.concept_id into @concept_id from concept_reference_map rmap, concept_name cn where cn.concept_id = rmap.concept_id and cn.name='Visit Diagnoses'
-            and cn.concept_name_type = 'FULLY_SPECIFIED' and cn.voided = 0 group by rmap.concept_id having count(*) = 1;
+            and cn.concept_name_type = 'FULLY_SPECIFIED' and cn.voided = 0 and cn.locale = 'en' group by rmap.concept_id having count(*) = 1;
 
-            UPDATE concept set version = 'BAHMNI-CORE-1.0' where concept_id = @concept_id;
+            UPDATE concept set version = 'BAHMNI-CORE-1.0' where concept_id = @concept_id and version is null ;
```

Comment:
How do you know that @concept_id has a value? Should there not be a check? although the precondition should guard against that. 
Also you can put the version clause in the above statement.

Reply:
we are setting the initial value to '0'. 
The preconditon has same query.